internal/api/courses/assignments/submissions/proxy/submit.go

```diff
@@ -0,0 +1,85 @@
+package proxy
+
+import (
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/config"
+	"github.com/edulinq/autograder/internal/grader"
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/model"
+	"github.com/edulinq/autograder/internal/timestamp"
+)
+
+type SubmitRequest struct {
+	core.APIRequestAssignmentContext
+	core.MinCourseRoleGrader
+	Files core.POSTFiles `json:"-"`
+
+	ProxyUser core.TargetCourseUser `json:"proxy-email"`
+	ProxyTime *timestamp.Timestamp  `json:"proxy-time"`
+
+	Message string `json:"message"`
+}
+
+type SubmitResponse struct {
+	FoundUser bool   `json:"found-user"`
+	Rejected  bool   `json:"rejected"`
+	Message   string `json:"message"`
+
+	GradingSuccess bool               `json:"grading-success"`
+	GradingInfo    *model.GradingInfo `json:"result"`
+}
+
+// Proxy submit an assignment submission to the autograder.
+func HandleSubmit(request *SubmitRequest) (*SubmitResponse, *core.APIError) {
+	response := SubmitResponse{}
+
+	if !request.ProxyUser.Found {
+		return &response, nil
+	}
+
+	// Proxy submissions are not subject to submission restrictions.
+	gradeOptions := grader.GradeOptions{
```

Comment:
Instead of constructing yourself,
you should probably get the default options and add your new options to it.